src/modules/function/glob.rs

```diff
@@ -0,0 +1,68 @@
+use std::mem::swap;
+
+use heraclitus_compiler::prelude::*;
+use itertools::Itertools;
+use crate::docs::module::DocumentationModule;
+use crate::modules::expression::expr::Expr;
+use crate::modules::types::{Type, Typed};
+use crate::translate::module::TranslateModule;
+use crate::utils::metadata::{ParserMetadata,TranslateMetadata};
+
+#[derive(Debug, Clone)]
+pub struct GlobInvocation {
+    args: Vec<Expr>,
+}
+
+impl Typed for GlobInvocation {
+    fn get_type(&self) -> Type {
+        Type::Array(Box::new(Type::Text))
+    }
+}
+
+impl SyntaxModule<ParserMetadata> for GlobInvocation {
+    syntax_name!("Glob Invocation");
+
+    fn new() -> Self {
+        GlobInvocation {
+            args: vec![],
+        }
+    }
+
+    fn parse(&mut self, meta: &mut ParserMetadata) -> SyntaxResult {
+        token(meta, "glob")?;
+        token(meta, "(")?;
+        let mut new_is_unquoted_ctx = true;
+        swap(&mut new_is_unquoted_ctx, &mut meta.context.is_unquoted_ctx);
```

Comment:
This pattern is dangerous, because the context value is not reverted to its previous value if the function does an early return, as it will if `syntax(meta, &mut arg)?` fails below.  However, the source code is full of instances of this, so we need a comprehensive fix.  I will create a separate issue for this.

Reply:
See #489.